library/alloc/src/vec/mod.rs

```diff
@@ -844,6 +847,16 @@ impl<T> Vec<T> {
         // SAFETY: A `Vec` always has a non-null pointer.
         (unsafe { NonNull::new_unchecked(ptr) }, len, capacity)
     }
+
+    /// Leaks the `Vec<T>` to be interned statically. This mut be done for all
+    /// `Vec<T>` created during compile time.
```

Comment:
It doesn't need to be done for all `Vec` created during compile-time -- only for those that you want to carry over to runtime. You can use intermediate `Vec` as scratch space or so during compile time that you never call this method on.